### PR TITLE
R doc 1799 license core assignment

### PR DESCRIPTION
**Pull request by morhilai**

### CODE REVIEW

File: Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+The easiest way to manage a RavenDB cluster is through the Management Studio's [Cluster View](../../studio/server/cluster/cluster-view).  
+All cluster operations are available through the GUI. However, sometimes you'd want to automate things. 
+
+In this section we will demonstrate of how to perform certain cluster operations programmatically using the REST API.
+In the first example we will show how to make a secure call using PowerShell, cURL and the RavenDB C# Client. The rest of the examples are cURL only.
+
+- [Add node to the cluster](../../server/clustering/cluster-api#add-node-to-the-cluster)
+- [Delete node from the cluster](../../server/clustering/cluster-api#delete-node-from-the-cluster)
+- [Promote a watcher node](../../server/clustering/cluster-api#promote-a-watcher-node)
```
Comment (garayx) [inline Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown:11]: there is no `promote-a-watcher-node` section, should point to `Promote a Node`, or rename Promote a Node
 to promote a watcher node

File: Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+- [Demote a watcher node](../../server/clustering/cluster-api#demote-a-watcher-node)
```
Comment (garayx) [inline Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown:12]: there is no `demote-a-watcher-node` section, should point to `Demote a Node
`, or rename Demote a Node
 to Demote a watcher node

File: Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+* To make a secure call, the Document Store must be supplied with the client certificate ([example](../../client-api/setting-up-authentication-and-authorization)).
+
+{CODE add_node_with_args@ClusterAPI.cs /}
+
+### Powershell
+
+{CODE-BLOCK:powershell}
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+$clientCert = Get-PfxCertificate -FilePath <path-to-pfx-cert>
+Invoke-WebRequest -Method Put -URI "http://<server-url>/admin/cluster/node?url=<new-node-url>&tag=<new-node-tag>&watcher=<is-watcher>&assignedCores=<assigned-cores> -Certificate $cert"
```
Comment (garayx) [inline Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown]: url should start with `https`

File: Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+{CODE-BLOCK/}
+
+### cURL
+
+{CODE-BLOCK: bash}
+curl -X PUT http://<server-url>/admin/cluster/node?url=<new-node-url>&tag=<node-tag>&watcher=<is-watcher>&assignedCores=<assigned-cores> --cert <path-to-pem-cert>
```
Comment (garayx) [inline Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown]: url should start with `https`

File: Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+Promoting a node can be done using an HTTP POST request to the `/admin/cluster/promote` 
+endpoint with the following argument:
+
+| Argument | Description | Required | Default |
+| - | - | - | - |
+| node-tag | The tag of the node to promote | true | -
+
+The POST request body should be empty.
+
+SecurityClearance: `Operator`
```
Comment (garayx) [inline Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown:85]:
```suggestion
SecurityClearance: `ClusterAdmin`
```

File: Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+
+Demoting a node can be done using an HTTP POST request to the `/admin/cluster/demote` endpoint with the following argument:
+
+| Argument | Description | Required | Default |
+| - | - | - | - |
+| node-tag | The tag of the node to demote | true | -
+
+The POST request body should be empty.
+
+SecurityClearance: `Operator`
```
Comment (garayx) [inline Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown:105]:
```suggestion
SecurityClearance: `ClusterAdmin`
```

File: Documentation/4.2/Raven.Documentation.Pages/server/clustering/rachis/cluster-topology.markdown [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+###Fetching the Topology
+Any client with [Valid User](../../security/authorization/security-clearance-and-permissions#user) privileges can fetch the topology from the server 
+by sending the following REST API call:  
+
+| Action | Method | URL |
+| - | - | - |
+| Fetch Topology | `GET` | /cluster/topology |
+
+###Modifying the Topology  
+With [Cluster Operator](../../security/authorization/security-clearance-and-permissions#operator) privileges,
```
Comment (garayx) [inline Documentation/4.2/Raven.Documentation.Pages/server/clustering/rachis/cluster-topology.markdown:29]: Cluster Operator and appropriate link

File: Documentation/4.2/Raven.Documentation.Pages/server/clustering/rachis/cluster-topology.markdown [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+
+{PANEL: Changing Node Type}
+
+Node types can be altered on the fly by using the following REST API calls:
+
+| Action | Method | URL |
+| - | - | - |
+| Promote Watcher to be a Member | `POST` | /admin/cluster/promote |
+| Demote Member to be a Watcher | `POST` | /admin/cluster/demote |
+| Force Elections | `POST` | /admin/cluster/reelect |
```
Comment (garayx) [inline Documentation/4.2/Raven.Documentation.Pages/server/clustering/rachis/cluster-topology.markdown:129]: add the requested privileges

File: Documentation/4.2/Raven.Documentation.Pages/studio/server/cluster/add-node-to-cluster.markdown [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+![Figure 0. Click to add a new node](images/cluster-add-node-0.png "Bootstrap Cluster")
+{PANEL/}
+
+{PANEL: Add Another Node to the Cluster}
+
+![Figure 1. Click to add a new node](images/cluster-add-node-1.png "Click to add a new node")
+
+<br/>
+![Figure 2. Adding a new node](images/cluster-add-node-2.png "Adding a new Node")
```
Comment (garayx) [inline Documentation/4.2/Raven.Documentation.Pages/studio/server/cluster/add-node-to-cluster.markdown:40]: the image is not expandable, and without title

File: Documentation/4.2/Raven.Documentation.Pages/studio/server/cluster/cluster-view.markdown [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+   See [Add node to cluster](add-node-to-cluster)
+2. **Demote**  
+   Demote a Member to be a Watcher  
+3. **Reassign Cores**  
+   Reassign the number of cores to be used by the server on this node  
+4. **Step Down**  
+   This option is available only on the Leader node  
+   A new voting process will be triggered and a new Leader will be elected  
+5. **Force Timeout**  
+   The default configuration for the RavenDB cluster is that each node expects to get a heartbeat from the cluster leader every 300 milliseconds.
```
Comment (garayx) [inline Documentation/4.2/Raven.Documentation.Pages/studio/server/cluster/cluster-view.markdown:73]:
```suggestion
   The default configuration for the RavenDB cluster is that each node expects to get a heartbeat from the cluster leader (default: every 300 milliseconds).  
```
add link to the configuration as well

File: Documentation/4.2/Raven.Documentation.Pages/studio/server/cluster/cluster-view.markdown [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+2. **Demote**  
+   Demote a Member to be a Watcher  
+3. **Reassign Cores**  
+   Reassign the number of cores to be used by the server on this node  
+4. **Step Down**  
+   This option is available only on the Leader node  
+   A new voting process will be triggered and a new Leader will be elected  
+5. **Force Timeout**  
+   The default configuration for the RavenDB cluster is that each node expects to get a heartbeat from the cluster leader every 300 milliseconds.  
+   Clicking 'Force Timeout' will trigger actions on the node as if it did Not hear from the Leader in this time period.
```
Comment (garayx) [inline Documentation/4.2/Raven.Documentation.Pages/studio/server/cluster/cluster-view.markdown:74]:
```suggestion
   Clicking 'Force Timeout' will trigger actions on the node as if it did not hear from the Leader in this time period.  
```

File: Documentation/4.2/Raven.Documentation.Pages/server/clustering/overview.markdown [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+* A node can be limited to a maximum number of cores it can use. This can be configured 
+[in the Studio](../../studio/server/cluster/cluster-view#reassign-cores).  
+{NOTE/}
+
+## Related Articles
+
+### Cluster in the Studio
+- [Cluster View](../../studio/server/cluster/cluster-view)
+- [Adding Node to Cluster](../../studio/server/cluster/add-node-to-cluster)  
+- [Cluster Observer Log](../../studio/server/cluster/cluster-observer)
```
Comment (garayx) [inline Documentation/4.2/Raven.Documentation.Pages/server/clustering/overview.markdown:75]: wrong link

File: Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown (hunk 2 above)
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown:12]: Fixed

File: Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown (hunk 1 above)
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown:11]: Fixed

File: Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown [hunk 13]
```diff
@@ -0,0 +1,156 @@
...
+| new-node-url | The address of the new node we want to add to the cluster | true | -
+| new-node-tag | 1-4 uppercase unicode letters | false | 'A' - 'Z' assigned by order of addition
+| is-watcher | Add the new node as a watcher | false | false
+| assigned-cores | The number of cores to assign to the new node | false | number of processors on the machine or the license limit (smallest)
+
+SecurityClearance: `Cluster Admin`
+
+### C# Client
+
+* To make a secure call, the Document Store must be supplied with the client certificate ([example](../../client-api/setting-up-authentication-and-authorization)).
```
[the remark this answers is not in the document]
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown:36]: Fixed

File: Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown (hunk 3 above)
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown]: Fixed

File: Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown (hunk 4 above)
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown]: Fixed

File: Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown (hunk 5 above)
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown:85]: Fixed

File: Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown (hunk 6 above)
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/server/clustering/cluster-api.markdown:105]: Fixed

File: Documentation/4.2/Raven.Documentation.Pages/studio/server/cluster/cluster-view.markdown (hunk 10 above)
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/studio/server/cluster/cluster-view.markdown]: Fixed

File: Documentation/4.2/Raven.Documentation.Pages/server/clustering/rachis/cluster-topology.markdown (hunk 7 above)
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/server/clustering/rachis/cluster-topology.markdown:29]: I don't understand, it already is Operator. Did you mean Cluster Admin?

File: Documentation/4.2/Raven.Documentation.Pages/server/clustering/rachis/cluster-topology.markdown [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+
+{NOTE: }
+
+#### Leader
+ 
+* The Leader makes sure that decisions are consistent at the cluster level, as long as a majority of the nodes are functioning and can talk to the Leader.  
+
+* For example, the decision to add a database to a node will be either accepted by the entire cluster (eventually) or fail to register altogether.  
+
+* [Raft Commands](../../../server/clustering/rachis/what-is-rachis#implementation-details) can't be accepted while there is no Leader or if the Leader is down.
```
[the remark this answers is not in the document]
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/server/clustering/rachis/cluster-topology.markdown:104]: Fixed

File: Documentation/4.2/Raven.Documentation.Pages/server/clustering/rachis/cluster-topology.markdown (hunk 8 above)
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/server/clustering/rachis/cluster-topology.markdown:129]: I added a column called "clearance" for the required security clearance
>Promote | | | ClusterAdmin
Demote | | | ClusterAdmin
Force Elections | | | Operator

Is this what you mean?

File: Documentation/4.2/Raven.Documentation.Pages/studio/server/cluster/cluster-view.markdown (hunk 11 above)
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/studio/server/cluster/cluster-view.markdown:74]: Fixed

File: Documentation/4.2/Raven.Documentation.Pages/server/clustering/overview.markdown (hunk 12 above)
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/server/clustering/overview.markdown:75]: Fixed

File: Documentation/4.2/Raven.Documentation.Pages/server/clustering/rachis/cluster-topology.markdown (hunk 7 above)
Comment (garayx) [reply Documentation/4.2/Raven.Documentation.Pages/server/clustering/rachis/cluster-topology.markdown:29]: yes

Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/server/clustering/rachis/cluster-topology.markdown:29]: Fixed

**Review by garayx — APPROVED** [review]

**Pull request merged**